fs/move.py

```diff
@@ -55,6 +59,21 @@ def move_file(
             resources (defaults to `False`).
 
     """
+    # optimization for moving files between different OSFS instances
+    if isinstance(src_fs, OSFS) and isinstance(dst_fs, OSFS):
+        try:
+            src_syspath = src_fs.getsyspath(src_path)
+            dst_syspath = dst_fs.getsyspath(dst_path)
+            common = commonpath([src_syspath, dst_syspath])
+            rel_src = frombase(common, src_syspath)
+            rel_dst = frombase(common, dst_syspath)
+            with open_fs(common, writeable=True) as base:
+                base.move(rel_src, rel_dst, preserve_time=preserve_time)
+            return
+        except (ValueError, errors.NoSysPath):
```

Comment:
If we know both FSes are OSFS instances, is it even possible to get a NoSysPath error?

Also, do we need to bother with calculating a commonpath, or can we just do `os.rename(src_syspath, dst_syspath)` ?


Hmmm, thinking about it some more, we could probably just do this if both `src_path` and `dst_path` have a SysPath, _regardless_ of what class their FSes are? And it probably makes sense to use the same logic that https://github.com/PyFilesystem/pyfilesystem2/blob/master/fs/base.py#L1134 does? (i.e. try to rename first, and then drop through to default behaviour). So perhaps this "optimisation" should go just above the `# Standard copy and delete` below?

Although no doubt there's intricacies that I've probably overlooked :laughing:

Reply:
>  we could probably just do this if both src_path and dst_path have a SysPath, regardless of what class their FSes are?

I tried that and had trouble with ZipFS

Reply:
Seems to me like a ZipFS shouldn't even have a SysPath? Perhaps that's a bug in ZipFS? :man_shrugging:

Reply:
Yes, in theory `ZipFS` shouldn't have a syspath, so that's a bug. But I concur with @lurch I think the easiest would be to write the code in the context (otherwise you lose the ability to optimize a copy between FS URLs) and simply do:
```python
if _src_fs is _dst_fs:
   ...
elif _src_fs.hassyspath(src_path) and _dst_fs.hassyspath(dst_path):
   ...
else:
   ...
```

Reply:
@lurch I think `os.rename` wouldn't work across drives on windows, but `shutil.move()` is an option. I decided against that because that would need some extra code for `preserve_time`...